include/circt/Dialect/Seq/SeqDialect.h

```diff
@@ -0,0 +1,36 @@
+//===- SeqDialect.h - Seq dialect declaration -------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines an Seq MLIR dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef CIRCT_DIALECT_SEQ_SEQDIALECT_H
+#define CIRCT_DIALECT_SEQ_SEQDIALECT_H
+
+#include "circt/Support/LLVM.h"
+#include "mlir/IR/Dialect.h"
+
+namespace circt {
+namespace seq {
+
+class SeqDialect : public Dialect {
+public:
+  explicit SeqDialect(MLIRContext *context);
+  ~SeqDialect();
+
+  static StringRef getDialectNamespace() { return "seq"; }
+
+  Operation *materializeConstant(OpBuilder &builder, Attribute value, Type type,
```

Comment:
I don't know if this is necessary as I'm not totally sure in what situations it's used. What's the rule here?

Reply:
It doesn't hurt to add this.  This is used by the constant folding logic, when `fold()` for an op returns an attribute.  This specifies the op to use to represent the constant (which should be rtl.constant).